src/Package/Impl/Repl/Commands/SourceFileHelper.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.R.Host.Client;
+using Microsoft.VisualStudio.R.Package.Shell;
+using Microsoft.VisualStudio.R.Package.Utilities;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.R.Package.Repl.Commands {
+    internal static class SourceFileHelper {
+        public static void SourceFiles(IEnumerable<string> files) {
+            VsAppShell.Current.AssertIsOnMainThread();
+
+            bool debugging = IsDebugging();
+            var operations = GetOperations();
+            var list = new List<string>();
+
+            Task.Run(async () => {
+                foreach (var f in files) {
+                    list.Add(await SessionUtilities.GetRShortenedPathNameAsync(f));
```

Comment:
User can change his current directory while this loop is running, so part of the paths will be pointing to the wrong place. And we actually don't need to call `GetRUserDirectoryAsync` on every step of the loop.

As for the helper methods, I think that for this type of extensions we should have them for `IRSessionEvaluation` rather than for `IRSession`.